### DISCUSSION TITLE
Problem with arrays in nested objects of base form

### DISCUSSION THREAD

**Discussion by Sindr0me** [react-hook-form/react-hook-form#6269] [Q&A]
Hi! Can anybody help me? 
I have next structure in my form: `{[key in string]: {  
  id: number;
  name: string;
  productId: number;
  coefficients: ActionPlanClientCoefficient[];
  comment: string;
  result: number;
}}`
ActionPlanClientCoefficient type is array of objects

In case on the screen I want to work with coefficients array. This works in my case with 'as' definition in defaultValue prop. I did it coz defaultValue prop is always undefined when im working with array types in form.

So can I work with arrays inside an objects and how is it possible with correct types?
Appears on migrtion from "react-hook-form": "^7.2.3" to "react-hook-form": "^7.12.2"
![image](https://user-images.githubusercontent.com/32894023/130088669-2395ebc3-a77d-41ec-8e20-50ea7574e22e.png)

**Comment by bluebill1049** [comment]
try with the latest `next` version.

**Reply by Sindr0me** [reply]
it still wrong with next version

![image](https://user-images.githubusercontent.com/32894023/130191442-e71d6dd0-6d7c-42c9-b506-876595bd99f0.png)

**Comment by Sindr0me** [comment]
i did the example here. As i can see the base form won't works with arrays coz it waits an objects therefore i can't set the 'number' in string type of name. Am I right that I can't use arrays - only objects? Or can I set number like '0' instead 'number' in type as you can see in my example? Is it like antipattern?

https://codesandbox.io/s/react-hook-form-typescript-forked-1cst8?file=/src/index.tsx

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/issues/6285
